action/protocol/staking/protocol.go

```diff
@@ -280,22 +280,26 @@ func (p *Protocol) CreateGenesisStates(
 // CreatePreStates updates state manager
 func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager) error {
 	var (
-		g                    = genesis.MustExtractGenesisContext(ctx)
-		blkCtx               = protocol.MustGetBlockCtx(ctx)
-		featureCtx           = protocol.MustGetFeatureCtx(ctx)
-		featureWithHeightCtx = protocol.MustGetFeatureWithHeightCtx(ctx)
+		g          = genesis.MustExtractGenesisContext(ctx)
+		blkCtx     = protocol.MustGetBlockCtx(ctx)
+		featureCtx = protocol.MustGetFeatureCtx(ctx)
 	)
 	if blkCtx.BlockHeight == g.GreenlandBlockHeight {
 		csr, err := ConstructBaseView(sm)
 		if err != nil {
 			return err
 		}
+		view := csr.BaseView()
 		if _, err = sm.PutState(csr.BaseView().bucketPool.total, protocol.NamespaceOption(_stakingNameSpace), protocol.KeyOption(_bucketPoolAddrKey)); err != nil {
 			return err
 		}
+		view.bucketPool.EnableSMStorage()
+		if err := sm.WriteView(_protocolID, view); err != nil {
```

Comment:
wondering is this necessarily needed? it's now base view constructed/read from `csr`, there's no changes to view/sm yet